lib/ts/ink_sock.cc

```diff
@@ -261,7 +261,7 @@ fd_read_line(int fd, char *s, int len)
 int
 close_socket(int s)
 {
-  return close(s);
+  return s < 0 ? -1 : close(s);
```

Comment:
Wouldn't close() return -1 and set EBADF anyways?

Reply:
Probably, but I think it's better to check. We've had some sporadic problems with double closes on file descriptors, which is where this comes from.

Reply:
I don't see how this would resolve the double close issue.  All it is doing is duplicating the check in another layer and hiding why the close failed by not setting errno.